### PR TITLE
[Exam] student exam detail

### PR DESCRIPTION
**Pull request by sjagla27**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Description

- Detail page for student exams. 
- Assessment and submission buttons are currently not implemented since there are no submissions or participations for the exam exercises yet.
- Download PDF button not implemented yet as the server side functionality does not exist at the moment

### Steps for Testing
1. Create an exam
2. Register a user for the exam (exam overview --> students)
3. Add at least one exercise group with at least on exercise to the exam (exam overview --> exercise groups)
4. Generate student exam (exam overview --> student exams)
5. Click on *View* next to the generated student exam

### Screenshots
grafik
The empty line will not be there once you don't have to manually set the exercises in your db.

### CODE REVIEW

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts [hunk 1]
```diff
@@ -10,14 +14,49 @@ import { StudentExamService } from 'app/exam/manage/student-exams/student-exam.s
...
+
+    loadAll() {
         this.courseId = Number(this.route.snapshot.paramMap.get('courseId'));
         this.route.data.subscribe(({ studentExam }) => (this.studentExam = studentExam));
+
+        this.courseService.find(this.courseId).subscribe((courseResponse) => {
+            this.course = courseResponse.body!;
+        });
+        // @ts-ignore
+        this.student = this.studentExam.user;
```
Comment (sascha11110) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts:37]: Don't ignore useful type warning :-D 

I think this should be `this.studentExam.student` or am I mistaken?

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts [hunk 2]
```diff
@@ -10,14 +14,49 @@ import { StudentExamService } from 'app/exam/manage/student-exams/student-exam.s
...
+    constructor(private route: ActivatedRoute, private studentExamService: StudentExamService, private courseService: CourseManagementService) {}
 
     /**
      * Initialize the courseId and studentExam
      */
     ngOnInit(): void {
+        this.loadAll();
+    }
+
+    loadAll() {
```
Comment (sascha11110) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts:29]: Please add documentation to all public methods or make them private (I think this one can be private)

Comment (sjagla27) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts:29]: ist now private

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts (hunk 1 above)
Comment (sjagla27) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts]: fixed

**Comment by Varnika-Tyagi** [comment]
Works as intended, on test server! Well done!

**Pull request merged**